### PR TITLE
Distributed PcaCov

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl_dpc.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+template <typename Float>
+result_t train_kernel_cov_impl<Float>::operator()(const descriptor_t& desc, const input_t& input) {
+    ONEDAL_ASSERT(input.get_data().has_data());
+    const auto data = input.get_data();
+
+    std::int64_t row_count = data.get_row_count();
+    auto rows_count_global = row_count;
+    //ONEDAL_ASSERT(data.get_column_count() > 0);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl_dpc.cpp:163]: Uncomment?

File: cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl_dpc.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    return std::make_tuple(corr, corr_event);
+}
+
+template <typename Float>
+auto compute_eigenvectors_on_host(sycl::queue& q,
+                                  pr::ndarray<Float, 2>&& corr,
+                                  std::int64_t component_count,
+                                  const dal::backend::event_vector& deps = {}) {
+    ONEDAL_PROFILER_TASK(compute_eigenvectors_on_host);
+    ONEDAL_ASSERT(corr.has_data());
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl_dpc.cpp:141]:
```suggestion
    ONEDAL_ASSERT(corr.has_mutable_data());
```

File: cpp/oneapi/dal/algo/pca/detail/train_ops.cpp [hunk 3]
```diff
@@ -21,26 +21,25 @@
...
+    template struct ONEDAL_EXPORT train_ops_dispatcher<dal::detail::spmd_host_policy, F, M, T>;
 
 INSTANTIATE(float, method::cov, task::dim_reduction)
 INSTANTIATE(float, method::svd, task::dim_reduction)
 INSTANTIATE(double, method::cov, task::dim_reduction)
 INSTANTIATE(double, method::svd, task::dim_reduction)
 
 } // namespace v1
-} // namespace oneapi::dal::pca::detail
+} // namespace oneapi::dal::pca::detail
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/detail/train_ops.cpp:45]: New line?

File: cpp/oneapi/dal/algo/pca/detail/train_ops_dpc.cpp [hunk 4]
```diff
@@ -22,27 +22,28 @@
...
+        train_ops_dispatcher<dal::detail::spmd_data_parallel_policy, F, M, T>;
 
 INSTANTIATE(float, method::cov, task::dim_reduction)
 INSTANTIATE(float, method::svd, task::dim_reduction)
 INSTANTIATE(double, method::cov, task::dim_reduction)
 INSTANTIATE(double, method::svd, task::dim_reduction)
 
 } // namespace v1
-} // namespace oneapi::dal::pca::detail
+} // namespace oneapi::dal::pca::detail
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/detail/train_ops_dpc.cpp:49]: new line?

File: cpp/oneapi/dal/algo/pca/test/fixture.hpp [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        const auto variances = result.get_variances();
+        const auto eigenvalues = result.get_eigenvalues();
+        const auto eigenvectors = result.get_eigenvectors();
+        return std::make_tuple(means, variances, eigenvalues, eigenvectors);
+    }
+};
+
+//using pca_types = COMBINE_TYPES((float, double), (pca::method::cov));
+
+} // namespace oneapi::dal::pca::test
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/test/fixture.hpp]: new line?

**Review by makart19 — APPROVED** [review]

File: cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl.hpp [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl.hpp:2]: 2022?

File: cpp/oneapi/dal/algo/pca/test/fixture.hpp [hunk 7]
```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/pca/test/fixture.hpp:2]: 2022?

**Review by amgrigoriev — APPROVED** [review]

**Pull request merged**